packages/test/src/test-integration-workflows.ts

```diff
@@ -216,6 +216,124 @@ test('Start of workflow is delayed', async (t) => {
   t.is(tsToMs(startDelay), 5678000);
 });
 
+export async function conflictId(): Promise<void> {
+  await workflow.condition(() => false);
+}
+
+test('Start of workflow respects workflow id conflict policy', async (t) => {
+  const { createWorker, taskQueue } = helpers(t);
+  const wfid = `${taskQueue}-` + randomUUID();
+  const client = t.context.env.client;
+
+  const worker = await createWorker();
+  await worker.runUntil(async () => {
+    const handle = await client.workflow.start(conflictId, {
+      taskQueue,
+      workflowId: wfid,
+    });
+    const handleWithRunId = client.workflow.getHandle(handle.workflowId, handle.firstExecutionRunId);
+
+    // Confirm another fails by default
+    const err = await t.throwsAsync(
+      client.workflow.start(conflictId, {
+        taskQueue,
+        workflowId: wfid,
+      }),
+      {
+        instanceOf: WorkflowExecutionAlreadyStartedError,
+      }
+    );
+
+    t.true(err instanceof WorkflowExecutionAlreadyStartedError && err.message === 'Workflow execution already started');
```

Comment:
Add `message` to the `throwsAsync()` checks above…

```
      {
        instanceOf: WorkflowExecutionAlreadyStartedError,
        message: 'Workflow execution already started',
      }
```

Reply:
Or you can even just not check for the error message. Given that it is an instance of the correct error class, that should be enough for this specific test.

Reply:
(same for all the tests below)

Reply:
Done